### ISSUE TITLE
MQTT Source wildcard issue: Invalid connect.source.converter.avro.schemas is not configured for topic

### ISSUE THREAD

**Issue by caevv** [lensesio/stream-reactor#559]
Version: 1.2.1
Kafka version: 2.1

Hi, I’m getting this error on a MQTT Source
```Caused by: org.apache.zookeeper.server.quorum.QuorumPeerConfig$ConfigException: Invalid connect.source.converter.avro.schemas is not configured for v1.mysite.studio/dev-123/computer/sensor/yoctopuce/gyroscope```

Config is this:
```
"connect.mqtt.kcql": "INSERT INTO spm_record SELECT * FROM v1.mysite.studio/+/computer/sensor/yoctopuce/gyroscope WITHCONVERTER=`com.datamountaineer.streamreactor.connect.converters.source.AvroConverter` WITHKEY(imo)", 
"connect.source.converter.avro.schemas":"v1.mysite.studio/+/computer/sensor/yoctopuce/gyroscope=/avroschema/spm_record.avro"
```

If I don't use wildcard it works.

Full error trace:

> [2019-03-19 15:14:01,593] ERROR Error handling message with id:1 on topic:v1.mysite.studio/dev-123/computer/sensor/yoctopuce/gyroscope (com.datamountaineer.streamreactor.connect.mqtt.source.MqttManager)
org.apache.zookeeper.server.quorum.QuorumPeerConfig$ConfigException: Invalid connect.source.converter.avro.schemas is not configured for v1.mysite.studio/dev-123/computer/sensor/yoctopuce/gyroscope
at com.datamountaineer.streamreactor.connect.converters.source.AvroConverter$$anonfun$1.apply(AvroConverter.scala:50)
at com.datamountaineer.streamreactor.connect.converters.source.AvroConverter$$anonfun$1.apply(AvroConverter.scala:50)
at scala.collection.MapLike$class.getOrElse(MapLike.scala:128)
at scala.collection.AbstractMap.getOrElse(Map.scala:59)
at com.datamountaineer.streamreactor.connect.converters.source.AvroConverter.convert(AvroConverter.scala:50)
at com.datamountaineer.streamreactor.connect.mqtt.source.MqttManager.messageArrived(MqttManager.scala:89)
at org.eclipse.paho.client.mqttv3.internal.CommsCallback.deliverMessage(CommsCallback.java:499)
at org.eclipse.paho.client.mqttv3.internal.CommsCallback.handleMessage(CommsCallback.java:402)
at org.eclipse.paho.client.mqttv3.internal.CommsCallback.run(CommsCallback.java:206)
at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
at java.util.concurrent.FutureTask.run(FutureTask.java:266)
at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
at java.lang.Thread.run(Thread.java:748)
[2019-03-19 15:14:01,629] WARN Connection lost. Re-connecting is set to true (com.datamountaineer.streamreactor.connect.mqtt.source.MqttManager)
MqttException (0) - org.apache.zookeeper.server.quorum.QuorumPeerConfig$ConfigException: Invalid connect.source.converter.avro.schemas is not configured for v1.mysite.studio/dev-123/computer/sensor/yoctopuce/gyroscope
at org.eclipse.paho.client.mqttv3.internal.CommsCallback.run(CommsCallback.java:220)
at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
at java.util.concurrent.FutureTask.run(FutureTask.java:266)
at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
at java.lang.Thread.run(Thread.java:748)
Caused by: org.apache.zookeeper.server.quorum.QuorumPeerConfig$ConfigException: Invalid connect.source.converter.avro.schemas is not configured for v1.mysite.studio/dev-123/computer/sensor/yoctopuce/gyroscope
at com.datamountaineer.streamreactor.connect.converters.source.AvroConverter$$anonfun$1.apply(AvroConverter.scala:50)
at com.datamountaineer.streamreactor.connect.converters.source.AvroConverter$$anonfun$1.apply(AvroConverter.scala:50)
at scala.collection.MapLike$class.getOrElse(MapLike.scala:128)
at scala.collection.AbstractMap.getOrElse(Map.scala:59)
at com.datamountaineer.streamreactor.connect.converters.source.AvroConverter.convert(AvroConverter.scala:50)
at com.datamountaineer.streamreactor.connect.mqtt.source.MqttManager.messageArrived(MqttManager.scala:89)
at org.eclipse.paho.client.mqttv3.internal.CommsCallback.deliverMessage(CommsCallback.java:499)
at org.eclipse.paho.client.mqttv3.internal.CommsCallback.handleMessage(CommsCallback.java:402)
at org.eclipse.paho.client.mqttv3.internal.CommsCallback.run(CommsCallback.java:206)
... 7 more

**Comment by brokenjacobs** [comment]
Is there any way to wildcard a schema?

**Comment by brokenjacobs** [comment]
Looking through the code there are a few ways this could be addressed. KCQL maps the topic name for each topic when a wildcard is used, kcql could leave the topic name as the wildcard name.

Another way would be to have the AvroConverter class understand wildcards instead of doing a ExistsOrElse on the map that contains the schemas. But this seems hard to do in a way that would work with other wildcard schemes outside of MQTT. 

In any case, I can't find a way to configure this in the current codebase and have it work.

**Comment by brokenjacobs** [comment]
KCQL supports WITHTARGET to dynamically choose a sink topic. Perhaps a keyword like WITHSOURCE to dynamically reduce multiple topics to one? Any thoughts on this stream reactor folks?

**Comment by andrewstevenson** [comment]
@brokenjacobs Sounds interesting but it would need an update to KCQL. Are you up for a PR?

**Comment by andrewstevenson** [comment]
@caevv this is be fixed one we migrate to scala 2.12 and is addressed in this branch

**Comment by andrewstevenson** [comment]
Fixed in #669
